master/buildbot/reporters/github.py

```diff
@@ -0,0 +1,158 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+
+"""Push events to github
+
+."""
+
+from datetime import datetime
+
+from buildbot import config
+from buildbot.util import service
+from buildbot.reporters import utils
+from buildbot.process.properties import Properties
+from buildbot.process.properties import Interpolate
+from buildbot.process.results import EXCEPTION
+from buildbot.process.results import FAILURE
+from buildbot.process.results import RETRY
+from buildbot.process.results import Results
+from buildbot.process.results import SUCCESS
+from buildbot.process.results import WARNINGS
+
+from twisted.internet import defer
+from twisted.internet import reactor
+from twisted.internet.protocol import ProcessProtocol
+from twisted.python import log
+
+try:
+    from txgithub.api import GithubApi as GitHubAPI
+except ImportError:
+    GitHubAPI = None
+
+class GitHubStatusPush(service.BuildbotService):
+
+    name = "GitHubStatusPush"
+
+    def reconfigService(self, token,
+            startDescription=None, endDescription=None,
+            context=None, baseURL=None, builders=None):
+        self.builders = builders
+
+        """
+        Token for GitHub API.
+        """
+        if not GitHubAPI:
+            config.error('GitHubStatus requires txgithub package installed')
+
+        self.context = context or Interpolate('buildbot/%(prop:buildername)s')
+        self.startDescription = startDescription or 'Build started.'
+        self.endDescription = endDescription or 'Build done.'
+
+        self.github = GitHubAPI(oauth2_token=token, baseURL=baseURL)
+
+        self.handledBuildIds = set()
+
+    @defer.inlineCallbacks
+    def startService(self):
+        yield service.BuildbotService.startService(self)
+        startConsuming = self.master.mq.startConsuming
+
+        self._buildUpdatedConsumer = yield startConsuming(
+            self.sendGitHubStatus,
+            ('builds', None, 'update'))
```

Comment:
Am 19.03.2016 12:03 nachm. schrieb "Pierre Tardy" <notifications@github.com
>:
>
> In master/buildbot/reporters/github.py:
>
> > +        self.context = context or
Interpolate('buildbot/%(prop:buildername)s')
> > +        self.startDescription = startDescription or 'Build started.'
> > +        self.endDescription = endDescription or 'Build done.'
> > +
> > +        self.github = GitHubAPI(oauth2_token=token, baseURL=baseURL)
> > +
> > +        self.handledBuildIds = set()
> > +
> > +    @defer.inlineCallbacks
> > +    def startService(self):
> > +        yield service.BuildbotService.startService(self)
> > +        startConsuming = self.master.mq.startConsuming
> > +
> > +        self._buildUpdatedConsumer = yield startConsuming(
> > +            self.sendGitHubStatus,
> > +            ('builds', None, 'update'))
>
> why didn't you use the "new" event?

For some reason or another, the new event didn't give me all the properties
in my case, all properties were empty.

>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub